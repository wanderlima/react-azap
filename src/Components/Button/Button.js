import React from 'react';
import styled, { css } from 'styled-components';
import defaultTheme from '../../theme';


const ButtonWrapper = styled.button`

	outline: none;
	cursor: pointer;
	background: none;
	padding: 0.5em 1em;
	color: ${props => props.theme.font.color};
	font-size: ${props => props.theme.font.size};
	font-family: ${props => props.theme.font.main};
	height: ${props => props.theme.button.minHeight};
	border: solid 0.2em ${props => props.theme.color.primary};
	border-radius: ${props => props.theme.button.borderRadius.min};

	:disabled{
		cursor: initial;
		color: ${props => props.theme.color.gray.main};
		background: ${props => props.theme.color.gray.light};
		border-color: ${props => props.theme.color.gray.main};
	}

`;


const Button = (props) => {
	return (
		<ButtonWrapper
			className="azap-button"
			{...props}
		>
			{props.children}
		</ButtonWrapper>
	);
}


export default Button;


Button.defaultProps = {
	theme: defaultTheme
}
