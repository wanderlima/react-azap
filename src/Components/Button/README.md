# Button

A component to blah blah...

``` js
<Button
	onClick={ () => { console.log('clicked') } }
	disabled={false}
>
	Click me
</Button>
```
