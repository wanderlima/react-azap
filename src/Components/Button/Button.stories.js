import React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import { boolean, text, withKnobs } from '@storybook/addon-knobs';
import { withReadme } from 'storybook-readme';

import { Button } from '../../components';
import README from './README.md';

const stories = storiesOf('Button', module);
stories.addDecorator(withReadme(README));
stories.addDecorator(withKnobs);


stories
	.add(
		'Default', () => (
			<Button
				onClick={ action('onClick') }
				disabled={ boolean('Disabled', false) }
			>
				{text('Label', 'Label')}
			</Button>
		)
	);
