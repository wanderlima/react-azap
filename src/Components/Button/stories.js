import React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import { boolean, text, withKnobs } from '@storybook/addon-knobs';
import { withDocs } from 'storybook-readme';

import { Button } from '../../Components';
import README from './README.md';

const stories = storiesOf('Button', module);
stories.addDecorator(withKnobs);

stories
	.add(
		'Default',
		withDocs(README, () => (
			<Button
				onClick={action('onClick')}
				disabled={boolean('Disabled', false)}
			>
				{text('Label', 'Button Label')}
			</Button>
		))
	);
