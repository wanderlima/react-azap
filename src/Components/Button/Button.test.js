import React from 'react';
import { shallow } from 'enzyme';

import Button from './Button';

describe('Testing Button component', () => {

	it('should render correctly', () => {
		const wrapper = shallow(
			<Button onClick={jest.fn}>
				<span>Label</span>
			</Button>
		);
		expect(wrapper).toMatchSnapshot();
	})

});
