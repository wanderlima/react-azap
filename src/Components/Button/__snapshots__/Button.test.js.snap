// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Button component should render correctly 1`] = `
<Button__ButtonWrapper
  className="azap-button"
  onClick={[Function]}
  theme={
    Object {
      "button": Object {
        "borderRadius": Object {
          "min": "1px",
        },
        "height": "2.6em",
      },
      "color": Object {
        "gray": Object {
          "dark": "#202020",
          "light": "#F7F7F7",
          "main": "#CCC",
        },
        "primary": "#FFEB3B",
        "text": Object {
          "main": "#202020",
        },
      },
      "font": Object {
        "color": "#202020",
        "main": "Source Sans Pro, sans-serif",
        "minHeigh": "1em",
        "monospace": "",
        "size": "1em",
      },
    }
  }
>
  <span>
    Label
  </span>
</Button__ButtonWrapper>
`;
